src/compiler/compile/render_dom/index.ts

```diff
@@ -333,8 +333,10 @@ export default function dom(
 		// $$props arg is still needed for unknown prop check
 		args.push(x`$$props`);
 	}
-
-	const has_create_fragment = component.compile_options.dev || block.has_content();
+	// fix: remove "component.compile_options.dev" condition, which
+	// will set has_create_fragment always be true in dev mode,
+  // inconsistent with the behavior in production mode.
+	const has_create_fragment = block.has_content();
```

Comment:
I don't think we should change here (and `src/compiler/compile/render_dom/Block.ts`).
Because `component.compile_options.dev` is Intentionally added by this commit.
https://github.com/sveltejs/svelte/pull/4501

And sometimes contributors modify compiled files directly for analysis.
At that time, It is useful if always the fragment is available.

On the other hand, such bug should detect by test.
Therefore personally I think we don't need to change here.

Reply:
Got it.